qiskit/pulse/instructions/instruction.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+``Instruction`` s are single events within a :py:class:`~qiskit.pulse.Schedule`, and can be
+used the same way as :py:class:`~qiskit.pulse.Schedule` s.
+
+Specific instructions should be instantiated rather than a generic ``Instruction``.
+
+For example::
+
+    duration = 10
+    channel = DriveChannel(0)
+    sched = Schedule()
+    sched += Delay(duration, channel)  # Delay is a specific subclass of Instruction
+"""
+import warnings
+
+from typing import Any, Tuple, List, Iterable, Callable, Optional, Union
+
+from qiskit.pulse.channels import Channel
+from qiskit.pulse.interfaces import ScheduleComponent
+from qiskit.pulse.schedule import Schedule
+from qiskit.pulse.timeslots import Interval, Timeslot, TimeslotCollection
+
+# pylint: disable=missing-return-doc
+
+
+class Instruction(ScheduleComponent):
```

Comment:
Should this be an ABC to prevent this from being instantiated?

Reply:
I was thinking about that, too, and I think yes. In which case `operands` should be an abstract property

Reply:
ABC worked, abstractmethods did not (yet)